apps/web/src/components/Feed/FeedEventData.tsx

```diff
@@ -25,62 +33,23 @@ type FeedEventDataNonRecursiveProps = {
   eventDataRef: FeedEventDataNonRecursiveFragment$key;
 };
 
-export function NonRecursiveFeedEventData({
+type getEventComponentProps = {
+  eventData: FeedEventDataNonRecursiveFragment$data;
+  isSubEvent?: boolean;
+  query: FeedEventDataNonRecursiveQueryFragment$data;
+  caption: string | null;
+  feedMode: FeedMode;
+  eventDbid: string;
+};
+
+const getEventComponent = ({
```

Comment:
@tbezman  The idea here is I want to wrap all the feed components in the `<StyledEvent/>` tag

Example:

```jsx
<StyledEvent isSubEvent={isSubEvent}>{eventComponent}</StyledEvent>
```

instead of


```jsx
  switch (eventData?.__typename) {
    case 'CollectionCreatedFeedEventData':
      return (
        <StyledEvent isSubEvent={isSubEvent}>
         <CollectionCreatedFeedEvent
           isSubEvent={isSubEvent}
           eventDataRef={eventData}
           queryRef={query}
         />
       </StyledEvent>
      );
    case 'CollectionUpdatedFeedEventData':
      return (
        <StyledEvent isSubEvent={isSubEvent}>
         <CollectionUpdatedFeedEvent
           isSubEvent={isSubEvent}
           eventDataRef={eventData}
           queryRef={query}
         />
       </StyledEvent>
      );
      
      ......
```

I'm open if you have simpler way to do it